### PR TITLE
Added build instructions to readme

### PR DESCRIPTION
**Pull request by devinbost**
Added additional requirements for building LangStream and added some simple instructions for fixing CI style issues.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,6 +180,19 @@ Requirements for building the project:
...
+Additionally, you will need to install into your Python env:
+* Black
+* Ruff
+* Build
+
+You can verify your requirements and run your first build like this:
+
+```bash
+mvn clean install -DskipTests
+mvn generate-sources
```
Comment (nicoloboschi) [inline README.md:194]: we have maven wrapper so you don't need to install your own maven

you can run 
```
./mvnw ...
```

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -199,3 +212,9 @@ or for all the services
 ```bash
 mini-langstream dev build
 ```
+If you run into issues with the build passing locally but failing the style checks in the CI, you can run:
+```bash
+mvn license:format
```
Comment (eolivelli) [inline README.md:217]: I prefer to suggest to use the standard Maven phase "generate-sources"
./mvn generate-sources

This way in  the future we can change the plugins without having to care about this docs

File: README.md (hunk 1 above)
Comment (eolivelli) [reply README.md:194]: generate-sources is a phase that is implied by "install", you don't need the second command

**Review by eolivelli — COMMENTED** [review]
Thanks for improving the developer experience.
I also left some minor comments

**Pull request closed without merging**